pkg/types/types.go

```diff
@@ -233,8 +233,12 @@ func MergeMetrics(metrics [][]Metric, replicaMismatchConfig cfg.RenderReplicaMis
 		MismatchedPoints int    `json:"mismatched_points"`
 	}
 	var mismatchedMetricReports []metricReport
+	var diffResolutionMetrics []string
 	for _, ms := range metricByNames {
 		m, stats := mergeMetrics(ms, replicaMismatchConfig)
+		if !isStepTimeMatching(ms) {
+			diffResolutionMetrics = append(diffResolutionMetrics, m.Name)
```

Comment:
This can result in a high number of allocations. I'm a bit concerned about the performance impact of this. BTW, using a metric would resolve this problem.

Reply:
I tested on live box and didn't see perf degradation